language_detector/main.py

```diff
@@ -1,8 +1,9 @@
-# -*- coding: utf-8 -*-
-
-"""This is the entry point of the program."""
-
 def detect_language(text, languages):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    counts = { lang['name']:0 for lang in languages }
+    words = text.split()
+    for w in words:
+        for lang in languages:
+            if w in lang['common_words']:
+                counts[lang['name']]+=1
+    return max((count, name) for name, count in counts.items())[1]
```

Comment:
Ah, whew. Thanks, was wondering if I completely misunderstood some key thing about how max worked.

Er, re your edit, I don't think it would return the longest but the lexically last... ie, the last one in alphabetical order.